include/irr/asset/CTexturePacker.h

```diff
@@ -0,0 +1,107 @@
+#ifndef __IRR_C_TEXTURE_PACKER_H_INCLUDED__
+#define __IRR_C_TEXTURE_PACKER_H_INCLUDED__
+
+#include "irr/asset/format/EFormat.h"
+#include "irr/asset/ICPUImage.h"
+#include "irr/core/alloc/GeneralpurposeAddressAllocator.h"
+
+namespace irr {
+namespace asset
+{
+
+class CTexturePacker
+{
+    using addr_alctr_t = core::GeneralpurposeAddressAllocator<uint32_t>;
+
+public:
+    struct STexOffset
+    {
+        uint32_t x, y;
```

Comment:
textures are not allowed to be > 32k

either make x,y, into uint16_t,

OR

pack the whole offset into `uint32_t` and then provide template functions (for different amount of bits allocated to X,Y and Layer coordinates)